### DISCUSSION TITLE
Is it possible to exclude helm charts based if they're disabled?

### DISCUSSION THREAD

**Discussion by evanrich** [renovatebot/renovate#12934] [Request Help]
Really new to renovate, trying to figure how to configure this to auto update a lot of things I have.  I'm stuck right now with updating gitlab dependency charts, like the runner... here's my renovate config.js:

```
module.exports = {
  platform: 'gitlab',
  endpoint: 'https://gitlab.nottheserveryourelookingfor.com/api/v4/',
  token: process.env.RENOVATE_TOKEN,

  repositories: [
    'group1/project1',
    'group2/project2',
  ],

  hostRules: [
    {
      hostType: 'docker',
      username: 'notsofast',
      password: process.env.DOCKER_HUB_PASSWORD,
    },
  ], 

  logLevel: 'info',

  requireConfig: true,
  onboarding: true,
  onboardingConfig: {
    extends: ['config:base'],
    prConcurrentLimit: 5,
  },

  enabledManagers: [
    'dockerfile',
    'helmfile',
    'pip_requirements',
    'terraform',
  ],
};
```

I'm using the [gitlab managed repo example](https://gitlab.com/gitlab-org/project-templates/cluster-management) to control gitlab in my cluster, and as part of that it has a series of helm charts that Renovate detects:
```
applications/apparmor/helmfile.yaml (helmfile)
applications/cert-manager/helmfile.yaml (helmfile)
applications/cilium/helmfile.yaml (helmfile)
applications/elastic-stack/helmfile.yaml (helmfile)
applications/falco/helmfile.yaml (helmfile)
applications/fluentd/helmfile.yaml (helmfile)
applications/gitlab-runner/helmfile.yaml (helmfile)
applications/ingress/helmfile.yaml (helmfile)
applications/prometheus/helmfile.yaml (helmfile)
applications/sentry/helmfile.yaml (helmfile)
applications/vault/helmfile.yaml (helmfile)
```

and as such, when renovate first runs, it wants to open PR's for all of them:
![image](https://user-images.githubusercontent.com/518320/144528393-e8883583-e756-4b7c-b918-090a62527518.png)

However, the only one I want currently installed, is the gitlab-runner chart, as is defined in its helmfile.yaml:
```
repositories:
- name: gitlab
  url: https://charts.gitlab.io

releases:
- name: runner
  namespace: gitlab-managed-apps
  chart: gitlab/gitlab-runner
  version: 0.35.0
  installed: true
  values:
    - values.yaml
```
installed: true

All the rest have "installed: false"

Is there a way to make renovate look for this "installed: false" and ignore the helmfile if that is set? or do i need to add something like:
```
"ignoreDeps": ["vault", "sentry", ...]
```

Also, it appears that these rules or package rules are global, is there a way to specify rules per package manager/repository?  Thanks!

**Comment by rarkins** [comment]
You should specify rules per-repository using the renovate.json "repo config" file committed to the repos.

You can configure per-manager rules using matchManagers within packageRules.

Renovate doesn't currently detect installed values in helm charts, so you would need to ignore them either by name, or by file. What you could do for less config is this:

- one package rule to disable all helm charts (matchManagers=helmfile, enabled=false)
- a second package rule to enable just the one you want (matchManagers=helmfile, matchPackageNames=?, enabled=true

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#packagerules
